azimuth/routers/v1/model_performance/outcome_count.py

```
@@ -57,10 +57,9 @@ def get_outcome_count_per_threshold(
     )
 
     if len(task_result) == 0:
-        # Non-Editable postprocessing
-        return []
+        raise HTTPException(HTTP_400_BAD_REQUEST, detail="Postprocessing is not editable")
```

I don't have a specific reason to change that behavior, I simply noticed it might make more sense to return an error than silently return an empty plot. What do you think?

---

The front end will never call this plot if post-processing is unavailable right? I think I would raise in the module though, not in the route?

---

Right. The user doesn't get a way to navigate to the page, but nothing prevents him to type the URL (or refresh the page with a different config). So it's not impossible, but it's not in the usual workflow.

The module used to return an empty response
https://github.com/ServiceNow/azimuth/blob/5b9f9693d0900bee8b90fbc7c3aa5f336ec17f39/azimuth/modules/model_performance/outcome_count.py#L187-L190
And I changed it to return no response (see my comment https://github.com/ServiceNow/azimuth/pull/444/files#r1109929478 about the change)
https://github.com/ServiceNow/azimuth/blob/1015ad301d07bdea3cbf7630e33624b31a95f6de/azimuth/modules/model_performance/outcome_count.py#L187-L190
but it doesn't seem to `raise`. Am I missing something?

Do you think the module should `raise`? And here we `except` and re-`raise` an `HTTPException`?

---

Here it is! What do you think? d5648c6b8eaf43f71461296a03a6e3a6b3c68e33

---

Yes, perfect :) Thanks!!